### PR TITLE
Refactor/thr momentum management

### LINKED ISSUE #304: Convert thrMomentumManagement to C++
**Describe your use case**
The ``thrMomentumManagement`` module will have new features, but it needs to be converted to C++ first.

**Describe alternatives solutions you've considered**
Maintained all functionality but the module is now written in C++.

**Additional context**
N/A.

### PR DESCRIPTION
**Pull request by joaogvcarneiro**
* **Tickets addressed:** #304 
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description

This PR recreates the ``thrMomentumManagement`` module in C++ for further development. The first commit adds the module files, the second adds the unit test and the last adds documentation.

## Verification

Test files were copied from the C module except for some minor adjustments.

## Documentation

Ported the old module documentation into ``rst`` format.

## Future work

This conversion is intended to be used in future additions.

### CODE REVIEW

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: :small_red_triangle_down: You can delete the constructor/destructor given the compiler will generate a default  for each.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp]: :small_red_triangle_down: "Error: " is redundant in these log strings as the `BSK_ERROR` toggle prepends "BSK_ERROR" already.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i]: :small_red_triangle_down: These `struct` directives are no longer needed and can be removed.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (patkenneally) [inline src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: :large_blue_circle: There are many redundant comments throughout the unit tests in the code base. We should seek to clean them out as we go.

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.h]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.cpp]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/thrMomentumManagementCpp.i]: Done.

File: src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py
Comment (joaogvcarneiro) [reply src/fswAlgorithms/attControl/thrMomentumManagementCpp/_UnitTest/test_thrMomentumManagementCpp.py]: Done.

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
